### PR TITLE
Adding Port: Residentvania

### PR DESCRIPTION
**Pull request by Damian2809**
# New Port for Residentvania

## Game Information
- **Title**: Residentvania
- **URL**: https://oco.itch.io/residentvania
### CFW Tests
Ensure your game has been tested on all major CFWs:
- [X] AmberELEC
- [X] ArkOS
- [X] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [X] 480x320
- [X] 640x480
- [X] 720x720 (RGB30)
- [X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/re8/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Notes
+
+Thanks to [OcO](https://oco.itch.io/) for making this wonderful demake
```
Comment (Slayer366) [inline ports/re8/README.md:3]: Suggestion to have the link point to the actual game's page

**Review by Slayer366 — CHANGES_REQUESTED** [review]

File: ports/re8/README.md (hunk 1 above)
Comment (Damian2809) [reply ports/re8/README.md]: Fixed

File: ports/re8/screenshot.jpg
Comment (cdeletre) [inline ports/re8/screenshot.jpg:1]: screenshot must be 640x480

File: ports/re8/Residentvania.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+cd $GAMEDIR
+
+# Check for file existence before trying to manipulate them:
+[ -f "./gamedata/data.win" ] && mv gamedata/data.win gamedata/game.droid
+[ -f "./gamedata/game.win" ] && mv gamedata/game.win gamedata/game.droid
+[ -f "./gamedata/game.unx" ] && mv gamedata/game.unx gamedata/game.droid
+
+$GPTOKEYB "gmloader" -xbox36ß &
```
Comment (cdeletre) [inline ports/re8/Residentvania.sh:36]:
```suggestion
$GPTOKEYB "gmloader" -xbox360 &
```

File: ports/re8/re8/re8.xdelta
Comment (cdeletre) [inline ports/re8/re8/re8.xdelta:1]: It seems that this file is not used so it can be removed ?

File: ports/re8/re8/libs/libzip.so.5
Comment (cdeletre) [inline ports/re8/re8/libs/libzip.so.5:1]: Please add the [license](https://github.com/PortsMaster/PortMaster-New/blob/main/ports/minidoom2/minidoom2/licences/LICENSE.libzip.txt) :)

File: ports/re8/re8/libs/libcrypto.so.1.1
Comment (cdeletre) [inline ports/re8/re8/libs/libcrypto.so.1.1:1]: Please add the [license](https://github.com/PortsMaster/PortMaster-New/blob/main/ports/minidoom2/minidoom2/licences/LICENSE.libcrypto.txt) :)

**Review by cdeletre — CHANGES_REQUESTED** [review]

File: ports/re8/Residentvania.sh (hunk 2 above)
Comment (Damian2809) [reply ports/re8/Residentvania.sh:36]: Ah fuck, i have fixed that but notepad++ just said nah

File: ports/re8/re8/re8.xdelta
Comment (Damian2809) [reply ports/re8/re8/re8.xdelta:1]: That file should be used but my .sh is apparently the wrong one, I will fix it later

File: ports/re8/Residentvania.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+fi
+
+source $controlfolder/control.txt
+
+export PORT_32BIT="Y"
+
+get_controls
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+$ESUDO chmod 666 /dev/tty0
```
Comment (Cebion) [inline ports/re8/Residentvania.sh:22]:
```suggestion
not used in the .sh anymore
```

File: ports/re8/Residentvania.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+GAMEDIR="/$directory/ports/re8"
+
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$LD_LIBRARY_PATH"
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR/gamedata/"
+export GMLOADER_PLATFORM="os_linux"
+
+# We log the execution of the script into log.txt
+exec > >(tee "$GAMEDIR/log.txt") 2>&1
```
Comment (Cebion) [inline ports/re8/Residentvania.sh:32]: old logging method

File: ports/re8/Residentvania.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+fi
+
+
+# Check for file existence before trying to manipulate them:
+[ -f "./gamedata/data.win" ] && mv gamedata/data.win gamedata/game.droid
+[ -f "./gamedata/game.win" ] && mv gamedata/game.win gamedata/game.droid
+[ -f "./gamedata/game.unx" ] && mv gamedata/game.unx gamedata/game.droid
+
+# Make sure uinput is accessible so we can make use of the gptokeyb controls
+$ESUDO chmod 666 /dev/uinput
```
Comment (Cebion) [inline ports/re8/Residentvania.sh:49]:
```suggestion
```

Comment (Cebion) [reply ports/re8/Residentvania.sh:49]: Not used anymore.
Check https://portmaster.games/packaging.html#gamemaker-studio-gmloader-example-launchscript for up to date script examples

File: ports/re8/port.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{
+  "version": 3,
+  "name": "re8.zip",
```
Comment (Cebion) [inline ports/re8/port.json:3]: why re8?
It's a demake of 8 but the game name is residentvania not resident evil 8.

File: ports/re8/screenshot.jpg
Comment (Cebion) [inline ports/re8/screenshot.jpg:1]: Screeshot has to be 640x480

**Review by Cebion — CHANGES_REQUESTED** [review]

**Review by cdeletre — APPROVED** [review]

**Review by Cebion — APPROVED** [review]

**Pull request merged**